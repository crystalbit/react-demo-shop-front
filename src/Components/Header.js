import React from 'react';
import { Link } from 'react-router-dom';
import { makeStyles } from '@material-ui/core/styles';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import Avatar from '@material-ui/core/Avatar';
import logo from '../images/ninja.gif';
import Strings from '../Helpers/Strings';
import LoginLogoffButton from './Auth/LoginLogoffButton';

const useStyles = makeStyles(theme => ({
  root: {
    marginBottom: 20,
    marginRight: -10
  },
  menuButton: {
    marginRight: theme.spacing(2),
  },
  toolbar: {
    display: 'flex',
    justifyContent: 'space-between'
  },
  logo: {
    width: theme.spacing(7),
    height: theme.spacing(7),
    marginTop: 10,
    marginBottom: 10
  },
}));

export default function(props) {
  const classes = useStyles();

  return(
    <div className={classes.root}>
      <AppBar position="static">
        <Toolbar className={classes.toolbar}>
          <Avatar alt={Strings.shop.name} src={logo} className={classes.logo} component={Link} to="/" />
          <Typography variant="h6">
            {Strings.shop.name}
          </Typography>
          <LoginLogoffButton
            loginItem={props.loginItem}
            updateLoginItem={props.updateLoginItem}
          />
        </Toolbar>
      </AppBar>
    </div>
  );
}
