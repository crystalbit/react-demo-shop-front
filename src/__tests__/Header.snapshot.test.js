import React from 'react';
import renderer from 'react-test-renderer';
import Header from '../Components/Header';

test('Header equals its snaphot', () => {
  const component = renderer.create(
    <Header />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
