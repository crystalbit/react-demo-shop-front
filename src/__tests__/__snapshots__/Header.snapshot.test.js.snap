// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header equals its snaphot 1`] = `
<div
  className="makeStyles-root-1"
>
  <header
    className="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular makeStyles-toolbar-3 MuiToolbar-gutters"
    >
      <div
        className="MuiAvatar-root MuiAvatar-circle makeStyles-logo-4"
      >
        <img
          alt="Ninja Pizza!"
          className="MuiAvatar-img"
          src="ninja.gif"
        />
      </div>
      <h6
        className="MuiTypography-root MuiTypography-h6"
      >
        Ninja Pizza!
      </h6>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Login
        </span>
      </button>
    </div>
  </header>
</div>
`;
